libc-top-half/musl/src/env/__init_tls.c

```diff
@@ -16,6 +16,36 @@
 volatile int __thread_list_lock;
 
 #ifndef __wasilibc_unmodified_upstream
+
+extern unsigned char __heap_base;
```

Comment:
Perhaps add a commend like `These symbols are generated by wasm-ld`  and perhaps mentions why two of them are weak and add a `TODO` to rely on them exclusively once we drop support for llvm < N?